pkg/rules/rules.go

```diff
@@ -0,0 +1,133 @@
+package rules
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/oklog/run"
+	"github.com/pkg/errors"
+
+	"github.com/prometheus/client_golang/prometheus"
+	prometheus_config "github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/notifier"
+	prom_rules "github.com/prometheus/prometheus/rules"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgclient"
+	"github.com/timescale/promscale/pkg/rules/adapters"
+)
+
+const (
+	DefaultQueueCapacity   = 10000
+	DefaultOutageTolerance = time.Hour
+	DefaultForGracePeriod  = time.Minute * 10
+	DefaultResendDelay     = time.Minute
+)
+
+type Options struct {
+	QueueCapacity   int
+	OutageTolerance time.Duration
+	ForGracePeriod  time.Duration
+	ResendDelay     time.Duration
+}
+
+type manager struct {
+	rulesManager     *prom_rules.Manager
+	notifierManager  *notifier.Manager
+	discoveryManager *discovery.Manager
+}
+
+func NewManager(ctx context.Context, r prometheus.Registerer, client *pgclient.Client, opts *Options) (*manager, error) {
+	discoveryManagerNotify := discovery.NewManager(ctx, log.GetLogger(), discovery.Name("notify"))
+
+	notifierManager := notifier.NewManager(&notifier.Options{
+		QueueCapacity: opts.QueueCapacity,
+		Registerer:    r,
+		Do:            do,
+	}, log.GetLogger())
+
+	// For the moment, we do not have any external UI url, hence we provide an empty one.
```

Comment:
Let's create an issue to make such a UI and update this

Reply:
https://github.com/timescale/o11y-team-applications/issues/196